File: src/main/java/com/example/demo/model/City.java
package com.example.demo.model;

import javax.persistence.*;
import java.util.List;
import java.util.Objects;

@Entity
public class City {
    @Id
    @GeneratedValue(generator = "UUID")
    @org.hibernate.annotations.GenericGenerator(name = "UUID", strategy = "org.hibernate.id.UUIDGenerator")
    private String id;
    private Integer queryCost;
    private Double latitude;
    private Double longitude;
    private String resolvedAddress;
    private String address;
    private String timezone;
    private Double tzoffset;

    @OneToMany(mappedBy = "city", fetch = FetchType.LAZY, cascade = CascadeType.ALL)
    private List<Day> days;

    @OneToOne(mappedBy = "city", fetch = FetchType.LAZY, cascade = CascadeType.ALL)
    private CurrentCondition currentConditions;

    public City() {
    }

    public City(String id, Integer queryCost, Double latitude, Double longitude, String resolvedAddress, String address, String timezone, Double tzoffset, List<Day> days, CurrentCondition currentConditions) {
        this.id = id;
        this.queryCost = queryCost;
        this.latitude = latitude;
        this.longitude = longitude;
        this.resolvedAddress = resolvedAddress;
        this.address = address;
        this.timezone = timezone;
        this.tzoffset = tzoffset;
        this.days = days;
        this.currentConditions = currentConditions;
    }

    public String getId() {
        return id;
    }

    public void setId(String id) {
        this.id = id;
    }

    public Integer getQueryCost() {
        return queryCost;
    }

    public void setQueryCost(Integer queryCost) {
        this.queryCost = queryCost;
    }

    public Double getLatitude() {
        return latitude;
    }

    public void setLatitude(Double latitude) {
        this.latitude = latitude;
    }

    public Double getLongitude() {
        return longitude;
    }

    public void setLongitude(Double longitude) {
        this.longitude = longitude;
    }

    public String getResolvedAddress() {
        return resolvedAddress;
    }

    public void setResolvedAddress(String resolvedAddress) {
        this.resolvedAddress = resolvedAddress;
    }

    public String getAddress() {
        return address;
    }

    public void setAddress(String address) {
        this.address = address;
    }

    public String getTimezone() {
        return timezone;
    }

    public void setTimezone(String timezone) {
        this.timezone = timezone;
    }

    public Double getTzoffset() {
        return tzoffset;
    }

    public void setTzoffset(Double tzoffset) {
        this.tzoffset = tzoffset;
    }

    public List<Day> getDays() {
        return days;
    }

    public void setDays(List<Day> days) {
        this.days = days;
    }

    public CurrentCondition getCurrentConditions() {
        return currentConditions;
    }

    public void setCurrentConditions(CurrentCondition currentConditions) {
        this.currentConditions = currentConditions;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (!(o instanceof City)) return false;
        City city = (City) o;
        return Objects.equals(getId(), city.getId()) &&
                Objects.equals(getQueryCost(), city.getQueryCost()) &&
                Objects.equals(getLatitude(), city.getLatitude()) &&
                Objects.equals(getLongitude(), city.getLongitude()) &&
                Objects.equals(getResolvedAddress(), city.getResolvedAddress()) &&
                Objects.equals(getAddress(), city.getAddress()) &&
                Objects.equals(getTimezone(), city.getTimezone()) &&
                Objects.equals(getTzoffset(), city.getTzoffset()) &&
                Objects.equals(getDays(), city.getDays()) &&
                Objects.equals(getCurrentConditions(), city.getCurrentConditions());
    }

    @Override
    public int hashCode() {
        return Objects.hash(getId(), getQueryCost(), getLatitude(), getLongitude(), getResolvedAddress(), getAddress(), getTimezone(), getTzoffset(), getDays(), getCurrentConditions());
    }
}

File: src/main/java/com/example/demo/model/City.kt
package com.example.demo.model


import javax.persistence.*
import org.hibernate.annotations.GenericGenerator

@Entity
data class City(
    @Id
    @GeneratedValue(generator = "UUID")
    @GenericGenerator(name = "UUID", strategy = "org.hibernate.id.UUIDGenerator")
    val id: String?= null,
    val queryCost:Int?= null,
    val latitude:Double?= null,
    val longitude:Double?= null,
    val resolvedAddress:String?= null,
    val address:String?= null,
    val timezone:String?= null,
    val tzoffset:Double?= null,

    @OneToMany(mappedBy = "city", fetch = FetchType.LAZY)
    val days: List<Day>?= null,

    @OneToOne(mappedBy = "city", fetch = FetchType.LAZY)
    val currentConditions: CurrentCondition?= null,

){
       override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (javaClass != other?.javaClass) return false

        other as City

        if (id != other.id) return false
        if (queryCost != other.queryCost) return false
        if (latitude != other.latitude) return false
        if (longitude != other.longitude) return false
        if (resolvedAddress != other.resolvedAddress) return false
        if (address != other.address) return false
        if (timezone != other.timezone) return false
        if (tzoffset != other.tzoffset) return false
        if (days != other.days) return false
        if (currentConditions != other.currentConditions) return false

        return true
    }

    override fun hashCode(): Int {
        var result = id?.hashCode() ?: 0
        result = 31 * result + (queryCost ?: 0)
        result = 31 * result + (latitude?.hashCode() ?: 0)
        result = 31 * result + (longitude?.hashCode() ?: 0)
        result = 31 * result + (resolvedAddress?.hashCode() ?: 0)
        result = 31 * result + (address?.hashCode() ?: 0)
        result = 31 * result + (timezone?.hashCode() ?: 0)
        result = 31 * result + (tzoffset?.hashCode() ?: 0)
        result = 31 * result + days.hashCode()
        result = 31 * result + currentConditions.hashCode()
        return result
    }

}
